lib/brainstem/dsl/configuration.rb

```diff
@@ -11,8 +11,13 @@ class Configuration
       # @params [Object] parent_configuration The parent configuration object
       #   which the new configuration object should use as a base.
       def initialize(parent_configuration = nil)
-        @parent_configuration = parent_configuration || ActiveSupport::HashWithIndifferentAccess.new
-        @storage = ActiveSupport::HashWithIndifferentAccess.new
+        @parent_configuration  = parent_configuration || ActiveSupport::HashWithIndifferentAccess.new
+        @storage               = ActiveSupport::HashWithIndifferentAccess.new
+
+        self.nonheritable_keys =
+          parent_configuration.respond_to?(:nonheritable_keys) &&
+          parent_configuration.nonheritable_keys ||
```

Comment:
So I just checked this. It works as expected in that direction... but there is also something unexpected to me here:

```ruby
[1] pry(main)> parent = Brainstem::DSL::Configuration.new
=> #<Brainstem::DSL::Configuration:0x007fb25a91ee98 @nonheritable_keys=[], @parent_configuration={}, @storage={}>
[2] pry(main)> child = Brainstem::DSL::Configuration.new(parent)
=> #<Brainstem::DSL::Configuration:0x007fb259b4d170
 @nonheritable_keys=[],
 @parent_configuration=#<Brainstem::DSL::Configuration:0x007fb25a91ee98 @nonheritable_keys=[], @parent_configuration={}, @storage={}>,
 @storage={}>
[3] pry(main)> child.nonheritable!(:a)
=> ["a"]
[5] pry(main)> parent.nonheritable!(:b)
=> ["a", "b"]
[6] pry(main)> parent.nonheritable!
ArgumentError: wrong number of arguments (0 for 1)
from /Users/mavenlink/workspace/brainstem/lib/brainstem/dsl/configuration.rb:67:in `nonheritable!'
[7] pry(main)> parent
=> #<Brainstem::DSL::Configuration:0x007fb25a91ee98 @nonheritable_keys=["a", "b"], @parent_configuration={}, @storage={}>
[8] pry(main)> child
=> #<Brainstem::DSL::Configuration:0x007fb259b4d170
 @nonheritable_keys=["a", "b"],
 @parent_configuration=#<Brainstem::DSL::Configuration:0x007fb25a91ee98 @nonheritable_keys=["a", "b"], @parent_configuration={}, @storage={}>,
 @storage={}>
[9] pry(main)> child.nonheritable_keys
=> ["a", "b"]
```

Look at `[5]`. That seems like counter-intuitive and incorrect behaviour to me (where the child also affects the parent). Does that seem wrong to you, or is that an intentional feature?